### PR TITLE
Migrate Ibeji Adapter from Freyja repo

### PR DESCRIPTION
**Pull request by wilyle**
Migrates the `IbejiAdapter` from the Freyja repo and makes updates as necessary

### CODE REVIEW

**Comment by wilyle** [comment]
This PR only makes changes that are necessary for the ibeji adapter to build and run properly. Other updates (such as config improvements as described in #4 ) will be made in future PRs

File: freyja_adapters/digital_twin/ibeji_adapter/README.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+The `res` directory contains two sample config files. The `ibeji_adapter_config.sample.json` will be copied to the build output, so you can edit this file to configure the adapter.
+
+### Common Configuration Fields
+
+Whether or not you are using Chariott, the following config values are required:
+
+- `service_type`: Specifies whether to call Ibeji directly or use Chariott's Service Discovery system to find it. Valid values are `"ChariottDiscoveryService"` and `"InVehicleDigitalTwinService"`.
+- `uri`: The URI for the selected service.
+- `max_retries`: The maximum number of times to retry failed attempts to communicate with the digital twin service.
+- `retry_interval_ms`: The duration between retries.
```
Comment (ashbeitz) [inline freyja_adapters/digital_twin/ibeji_adapter/README.md:16]: .. retries in milliseconds.

File: freyja_adapters/digital_twin/ibeji_adapter/README.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+## Configuration
+
+The `res` directory contains two sample config files. The `ibeji_adapter_config.sample.json` will be copied to the build output, so you can edit this file to configure the adapter.
+
+### Common Configuration Fields
+
+Whether or not you are using Chariott, the following config values are required:
+
+- `service_type`: Specifies whether to call Ibeji directly or use Chariott's Service Discovery system to find it. Valid values are `"ChariottDiscoveryService"` and `"InVehicleDigitalTwinService"`.
```
Comment (ashbeitz) [inline freyja_adapters/digital_twin/ibeji_adapter/README.md:13]: I know that you are just copying this, so no need to change it, but you may want to rethink the name of this in the future.

I see that there are two options for locating the In-Vehicle Digital Twin, statically via an uri in the config or dynamically at runtime via Chariott's discovery service. service_type does not seem an appropriate term to use to represent these two options.

**Review by ashbeitz — COMMENTED** [review]
A few comments.

File: freyja_adapters/digital_twin/ibeji_adapter/README.md (hunk 2 above)
Comment (wilyle) [reply freyja_adapters/digital_twin/ibeji_adapter/README.md:13]: I agree. I will address this in #4 which will improve the config for the adapters here in the same way as I've been doing in the Freyja repo. Perhaps something like `service_discovery_mode`?

Comment (ashbeitz) [reply freyja_adapters/digital_twin/ibeji_adapter/README.md:13]: Yes, much better.

**Review by ashbeitz — APPROVED** [review]
Approved.

**Pull request merged**